docs/core/extensions/logger-message-generator.md

```diff
@@ -0,0 +1,320 @@
+    "{OriginalFormat}": "Fixed message"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Warning",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Information",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 20,
+  "LogLevel": "Critical",
+  "Category": "LoggingExample",
+  "Message": "Value is 1.234568E+004",
+  "State": {
+    "Message": "Value is 1.234568E+004",
+    "value": 12345.6789,
+    "{OriginalFormat}": "Value is {value:E}"
+  }
+}
+```
+
+### Benefits with the source generator approach:
+
+- Allows the logging structure to be preserved and enables the exact format syntax required by https://messagetemplates.org/
+- Allows supplying alternative names for the holes and using format specifiers.
+- Allows to pass all of original data as-is without any complication around how it's stored prior to something being done with it other than creating a string.
+- Provides logging-specific diagnostics, e.g. it emits warnings for duplicate event ids.
+
+#### Benefits compared to using LoggerMessage.Define directly
+
+- Shorter and simpler syntax than current approach
+- Guided developer experience - the generator gives warnings to help developers do the right thing
+- Support for an arbitrary # of logging parameters. current approach tops out at 6
+- Support for dynamic log level, current approach doesn't support this
+
+### Known issues/limitations
+
+The first version of the generator uses `LoggerMessage.Define` under the hood where possible. The issues below track potential improvements to make on `Define` APIs to allow better support in the source generator.
+
+- _Support for an arbitrary # of logging parameters and dynamic logging_
+
+Currently maximum number of parameters allowed via `LoggerMessage.Define` (Refer to [dotnet/runtime#50913](https://github.com/dotnet/runtime/issues/50913)) is 6. To mitigate this either the source generator would need to fallback to another approach, or we would need to add more `Define` overloads for the generator to use. The issue also tracks adding APIs to allow `Define` with supporting dynamic logging as well.
+
+- _More robust support for handling message template parameters._
+
+The code generation using `LoggerMessage.Define` does not currently handle message templates when number of parameters do not match the specified template parameters even if the parameters used in the template are the same (Refer to [dotnet/runtime#51054](https://github.com/dotnet/runtime/issues/51054])). This could be something that the source generator supports as it evolves further.
```

Comment:
```suggestion
The code generation using `LoggerMessage.Define` does not currently handle message templates when number of parameters do not match the specified template parameters even if the parameters used in the template are the same (Refer to [dotnet/runtime#51054](https://github.com/dotnet/runtime/issues/51054)). This could be something that the source generator supports as it evolves further.
```
In the link-address a `]` too much.